src/results.jl

```diff
@@ -0,0 +1,130 @@
+"""
+   InternalIntegralResult
+
+Result of computing the internal integral I(phi, beta1, beta2)
+- value: the computed integral value
+- error: error estimate (Nothing if not available)
+"""
+struct InternalIntegralResult{T <: Number}
+    value::T
+    error::T
+end
+
+# Constructor for cases without error estimate
+InternalIntegralResult(value::T) where {T} = InternalIntegralResult(value, T(NaN))
```

Comment:
According to offline discussion you need to promote (value, error) and handle the case of `T::Int` specially.